src/crypter.rs

```diff
@@ -764,16 +789,19 @@ mod read {
     pub fn block(
         input: &[u8],
         output: &mut Vec<u8>,
-        stream: &mut Stream<Pull>,
+        stream: &mut PullStream,
         message_length: usize,
     ) -> Result<ReadStatus<bool>, SaltlickError> {
         let block_len = message_length + ABYTES;
         if input.len() < block_len {
             return Ok(ReadStatus::Incomplete(block_len - input.len()));
         }
+        output.clear();
+        output.reserve(input.len());
+        std::io::copy(&mut input.as_ref(), output).map_err(|_| SaltlickError::DecryptionFailure)?;
         let tag = stream
-            .pull_to_vec(&input[..block_len], None, output)
```

Comment:
Is it possible for this function to be called when `input.len() > block_len`? It looks like this now uses the full `input` instead of the first `block_len` bytes.